src/nlattr.rs

```diff
@@ -311,4 +312,28 @@ mod test {
         let nlattr_deserialized = Nlattr::<CtrlAttr, u16>::deserialize(&mut reader).unwrap();
         assert_eq!(nlattr_deserialized, nlattr_desired_deserialized);
     }
+
+    #[test]
+    fn test_nl_len_after_adding_nested_attributes() {
+        let create_aligned_attr = ||
+            Nlattr::new(None, CtrlAttr::Unspec, vec![1, 2, 3, 4]).unwrap();
+
+        let create_unaligned_attr = ||
+            Nlattr::new(None, CtrlAttr::Unspec, vec![1]).unwrap();
+
+        let mut nlattr = Nlattr::new::<Vec<u8>>(None, CtrlAttr::Unspec, vec![]).unwrap();
+        assert_eq!(nlattr.size(), 4);
+
+        nlattr.add_nested_attribute(create_aligned_attr()).unwrap();
+        assert_eq!(nlattr.size(), 12);
+
+        nlattr.add_nested_attribute(create_unaligned_attr()).unwrap();
+        // Commented out because the following fails at the moment.
+        // assert_eq!(nlattr.size(), 17);
```

Comment:
So I expect this to be `17`. Do you John?

I don't think the Netlink specification outlines what the length field of an outer attribute should be when it contains inner attributes that are padded. Ex: If an inner attribute has length `5`, I expect an outer attribute (with only that singular inner attribute) to have length `9`. But maybe the outer attribute having `12` is also an okay answer.

Reply:
A better thing to test might be this. Make sure the outer attribute `size()` is aligned and the inner attribute `size()` is not aligned as long as it contains no nested attributes but only a serialized Rust primitive. Does that make sense? I posted a longer explanation in the main thread on this PR.

Reply:
Kind of. Are you saying a new/different test that contains the following?

```rust
let nlattr = Nlattr::new(None, CtrlAttr::Unspec, vec![1]).unwrap();
assert_eq!(nlattr.size(), 5);
assert_eq!(nlattr.asize(), 8);
```

Reply:
More what I was thinking is this:
```rust
let mut nlattr = Nlattr::new(None, CtrlAttr::Unspec, vec![]).unwrap();
let subattr = Nlattr::new(None, CtrlAttr::Unspec, 5u16).unwrap();
nlattr.add_nested_attribute(subattr).unwrap();
assert_eq!(subattr.size(), 6);
assert_eq!(nlattr.size(), 10);
```

I would expect it to work this way because when attributes are not being nested, the length is provided without padding. As soon as we start nesting them, each attribute is added with padding leaving the `.size()`/`.asize()` for an attribute with nested attributes 4 byte aligned.